include/unifex/complete_adapt_stream.hpp

```diff
@@ -0,0 +1,52 @@
+// (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+
+#pragma once
+
+#include <unifex/stream_concepts.hpp>
+
+#include <functional>
+#include <type_traits>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _complete_adapt {
+template <typename Stream, typename AdaptFunc>
+struct _stream {
+  struct type;
+};
+template <typename Stream, typename AdaptFunc>
+using stream =
+    typename _stream<remove_cvref_t<Stream>, remove_cvref_t<AdaptFunc>>::type;
+
+template <typename Stream, typename AdaptFunc>
+struct _stream<Stream, AdaptFunc>::type {
+  Stream innerStream_;
+  AdaptFunc adapter_;
```

Comment:
These would likely benefit from being annotated as [[no_unique_address]] since both types could in theory be memberless.